tests/reboot/reboot_test.go

```diff
@@ -168,11 +168,22 @@ var _ = Describe("{ReallocateSharedMount}", func() {
 								TimeBeforeRetry: defaultCommandRetry,
 							},
 						})
+						Expect(err).NotTo(HaveOccurred())
 
 						// as we keep the storage driver down on node until we check if the volume, we wait a minute for
 						// reboot to occur then we force driver to refresh endpoint to pick another storage node which is up
-						logrus.Info("wait for 1 minute for node reboot")
+						logrus.Infof("wait for %v for node reboot", defaultCommandTimeout)
 						time.Sleep(defaultCommandTimeout)
+
+						// Start NFS server to avoid pods stuck in terminating state (PWX-24274)
+						err = Inst().N.Systemctl(*n, "nfs-server.service", node.SystemctlOpts{
```

Comment:
Good question. See the analysis in the PWX ticket for more info. But to answer your questions:

- We delay the NFS server startup on purpose to avoid I/O errors for sharedv4-svc apps. Besides, the node could be shut down (instead of reboot). Starting the NFS server is not going to help there since the node is powered off.

- The problem seems to occur for the bonnie++ app which seems to enter and exit D state frequently even in a normal operation. That is why it is vulnerable to this problem. We have not seen the same problem with any other apps that use sharedv4 volumes.

One thing we can try in future is to start the NFS server if PX is down on the node for an extended period of time. This current change allows us to move forward with the bonnie++ job and catch other bugs in the meantime.

Reply:
Ah ok makes sense :+1: